src/main/java/edu/kit/compiler/semantic/SemanticException.java

```diff
@@ -7,8 +7,22 @@
 
 public class SemanticException extends CompilerException {
 
+    private final SourceLocation sourceLocation;
+
+    /**
+     * This constructor should be avoided. Implementation needs to switch to the one with source location once we
+     * get the error location in our AST.
```

Comment:
The error location is already in our AST, isn't it? `AstNode` has `line` and `column` as fields.